apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsUuidControl.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useMemo } from "react"
+import {
+  ControlProps,
+  RankedTester,
+  rankWith,
+  schemaMatches,
+} from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+
+export const jsonFormsUuidControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.UuidControl,
+  schemaMatches((schema) => schema.format === "uuid"),
+)
+
+export const JsonFormsUuidControl = ({
+  data,
+  handleChange,
+  path,
+}: ControlProps) => {
+  const uuid = useMemo(() => data || crypto.randomUUID(), [path])
```

Comment:
question: why is dependency `[path]` when its not being used and the variable being used is `data`?

Reply:
i derped, removed; shouldn't need it since `uuid` is already based off `path`